dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/filters/DatacenterIdListDatacentersFilter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.filters;
+
+import com.google.common.base.Supplier;
+import org.jclouds.http.HttpException;
+import org.jclouds.http.HttpRequest;
+import org.jclouds.http.HttpRequestFilter;
+import org.jclouds.location.Zone;
+
+import javax.inject.Inject;
+import java.util.Set;
+
+/**
+ * Adds set of Datacenter IDs as set in jclouds.zones JVM property.
+ */
+public class DatacenterIdListDatacentersFilter implements HttpRequestFilter {
+   @Inject
+   @Zone
```

Comment:
Good Q - we were having a discussion about this ourselves the other day. Your input would be appreciated @nacx!

DD has a number of 'Geos', each has it's own cloud control API endpoint, and consists of 1 or more datacenters. At present we see our provider as being for a single Geo (as it's the simplest thing to do to start with), but maybe we should aim to have a single provider to cover them all in time. There is no global API endpoint, so when creating a compute service instance we'd have to call out to multiple Geos to find out where the user is based (their home Geo) and which 'shaddow' Geos they also have access to. We do have an API that can be called on any Geo to find a users home Geo that does this fan-out, but I'm not sure if it's publicly published or not.

Anyway, to answer your Q I think the mapping should be:
Region -> Geo (only 1 for now - we prob need to do something to model this?)
Zone -> Datacenter

Does that sound ok?